### DISCUSSION TITLE
Continuance?

### DISCUSSION THREAD

**Discussion by MichaelMartinez** [AntonOsika/gpt-engineer#124] [Q&A]
I am attempting to work with gpt-engineer in creating a flask app with a svelte frontend. I get through most of the python when an error or token limit manifests and its done.

Is there a way to have it start where it left off?

Some thoughts and ideas:

1. as a prototype, I am not super interested in unit tests out of the gate and they eat up a lot of tokens. I tried turning them off in STEPS and they seem to be hardcoded thus erroring on that aspect.
2. Use the DB to "keep score" and find a way to re-run the generation from a point in time
3. Make the STEPS more configurable thus allowing one to choose how to generate the project in the order they choose
4. Reflection of some kind where it can look at the code it has generated. User can have it re-parse the prompt or point to a different prompt where it can continue.

**Comment by AntonOsika** [comment]
Did you check

`scripts/rerun_edited_message_logs.py`?

**Reply by clickbrain** [reply]
Can we use that to feed it new information like errors we encounter running the previous version? UI am having some luck with putting errors into main_prompt on subsequent runs, but have had to repeat numerous times to get it to actually fix code syntax errors and a few times couldn't get it to do anything at all.

**Comment by RareMojo** [comment]
Wanted to say that I still feel like respec/improving using its previous code output feels like a colossal waste of tokens so far. Sometimes, it makes a change or two, and they usually make no difference or cause more problems.

At the moment, I decided to try and implement the missing `use_feedback` step. (Might be fully implemented on main now in its own way..)

You fill in a second prompt called `feedback`. This is you answering to the `review` identity.

This method seems to work best with one error, goal, or change at a time.

Example: I want the snake to increase  movement speed each time the score increases. 

Details:

1. `use_feedback` example pseudo
```
First,
System -> input['main_prompt']
User -> input['all_output.txt']
User -> input['feedback']
Identity -> identity['review']
Then,
Next -> identity['fix_code']
```

Prompts:

1. `Review` identity example
```
I aim to generate the most accurate and efficient code. I would greatly appreciate it if you could review the output of the code I've produced. Did it function as expected? Were the results in line with your requirements?
``` 

2. `Feedback` from project input example

```
My classic arcade game 'Snake' written in Python is producing an error about 'missing assignment' whenever I try to launch it.
```

My experiments favor this form of improvement over a literal respec and improve on its own. I do not have a gigantic sample size, though, nor do I have the privilege of using GPT-4. 

This has been my testing and experience with this idea. I'd love to see what other ideas are out there!

**Comment by diulama** [comment]
Hi @AntonOsika 

Newbie here, what argument should we be passing for `message_path` in scripts/rerun_edited_message_logs.py? Thanks!
